From 0b5f63c7224cb5c9a00397f0e0ef4a16aa58e516 Mon Sep 17 00:00:00 2001
From: Yuichi Kusakabe <yuichi.kusakabe@jp.fujitsu.com>
Date: Fri, 9 Jun 2017 20:43:06 +0900
Subject: [PATCH 1/3] Add hibernation image area

0x40000000 <-> 0x77FFFFFF : kernel
0x78000000 <-> 0x7FFFFFFF : hibernation image area
and Enable sdhi DMA support

Signed-off-by: Yuichi Kusakabe <yuichi.kusakabe@jp.fujitsu.com>
---
 arch/arm/include/asm/arch-rmobile/r8a7791.h |  3 +++
 arch/arm/include/asm/armv7.h                |  3 +++
 include/configs/porter.h                    | 24 ++++++++++++------------
 3 files changed, 18 insertions(+), 12 deletions(-)

diff --git a/arch/arm/include/asm/arch-rmobile/r8a7791.h b/arch/arm/include/asm/arch-rmobile/r8a7791.h
index c964f13..9e08da4 100644
--- a/arch/arm/include/asm/arch-rmobile/r8a7791.h
+++ b/arch/arm/include/asm/arch-rmobile/r8a7791.h
@@ -147,6 +147,8 @@
 
 #define DBSC3_0_DBADJ2		0xE67900C8
 #define DBSC3_1_DBADJ2		0xE67A00C8
+#define DBSC3_0_DBCALTR		0xE67900F8
+#define DBSC3_1_DBCALTR		0xE67A00F8
 
 #define CCI_400_MAXOT_1		0xF0091110
 #define CCI_400_MAXOT_2		0xF0092110
@@ -154,6 +156,7 @@
 #define CCI_400_QOSCNTL_2	0xF009210C
 
 #define	MXI_BASE		0xFE960000
+#define	MXI_VIN_QOS		0xFE96020C
 #define	MXI_QOS_BASE		0xFE960300
 
 #define SYS_AXI_SYX64TO128_BASE	0xFF800300
diff --git a/arch/arm/include/asm/armv7.h b/arch/arm/include/asm/armv7.h
index aad5bf7..0d4d612 100644
--- a/arch/arm/include/asm/armv7.h
+++ b/arch/arm/include/asm/armv7.h
@@ -31,6 +31,9 @@
 #define MIDR_CORTEX_A9_R1P3	0x411FC093
 #define MIDR_CORTEX_A9_R2P10	0x412FC09A
 
+/* valid bits in CBAR register / PERIPHBASE value */
+#define CBAR_MASK		0xFFFF8000
+
 /* Cortex-A15 revisions */
 #define MIDR_CORTEX_A15_R0P0	0x410FC0F0
 
diff --git a/include/configs/porter.h b/include/configs/porter.h
index 5567c7c..f652bab 100644
--- a/include/configs/porter.h
+++ b/include/configs/porter.h
@@ -91,14 +91,14 @@
 
 /* MEMORY */
 #define PORTER_SDRAM_BASE	0x40000000
-#define PORTER_SDRAM_SIZE	0x48000000
+#define PORTER_SDRAM_SIZE	0x80000000
 #define PORTER_UBOOT_SDRAM_SIZE	0x40000000
 
 #define CONFIG_SYS_LONGHELP
 #define CONFIG_SYS_PROMPT		"=> "
-#define CONFIG_SYS_CBSIZE		512
-#define CONFIG_SYS_PBSIZE		512
-#define CONFIG_SYS_MAXARGS		32
+#define CONFIG_SYS_CBSIZE		256
+#define CONFIG_SYS_PBSIZE		256
+#define CONFIG_SYS_MAXARGS		16
 #define CONFIG_SYS_BARGSIZE		512
 #define CONFIG_SYS_BAUDRATE_TABLE	{ 38400, 115200 }
 
@@ -205,15 +205,15 @@
 #define CONFIG_USB_HOST_ETHER	/* Enable USB Ethernet adapters */
 #define CONFIG_USB_ETHER_ASIX	/* Asix, or whatever driver(s) you want */
 
-#define CONFIG_ARMV7_LPAE               /* 64-bit MMU descriptors */
-#define CONFIG_SYS_ARM_CACHE_WRITEALLOC /* Make memory operations faster */
-
-#define CONFIG_SYS_ARCH_TIMER           /* Init arch timer */
-#define CONFIG_VE_ENABLED               /* Virtualization Extensions are enabled*/
-#define CONFIG_SYS_HZ_CLOCK             CONFIG_SYS_CLK_FREQ
+#define CONFIG_ARMV7_LPAE		/* 64-bit MMU descriptors */
+#define CONFIG_SYS_ARM_CACHE_WRITEALLOC	/* Make memory operations faster */
+#define CONFIG_SYS_ARCH_TIMER	/* Init arch timer */
+#define CONFIG_SYS_HZ_CLOCK    CONFIG_SYS_CLK_FREQ
 
+#if 1
 #define CONFIG_SH_DMA
-#define CONFIG_SH_SYS_DMAL_BASE         0xE6700000
-#define CONFIG_SH_SYS_DMAL_NCH          15
+#define CONFIG_SH_SYS_DMAL_BASE		0xE6700000
+#define CONFIG_SH_SYS_DMAL_NCH		15
+#endif
 
 #endif	/* __PORTER_H */
-- 
1.8.3.1

